src/wordlift/mappings/class-mappings-dbo.php

```diff
@@ -55,27 +55,37 @@ public function get_mappings() {
 	public function get_active_mappings() {
 		$mapping_table_name = $this->wpdb->prefix . WL_MAPPING_TABLE_NAME;
 
-		return $this->wpdb->get_results(
+		$mappings = $this->wpdb->get_results(
 			"SELECT * FROM $mapping_table_name WHERE mapping_status = 'active'",
 			ARRAY_A
 		);
+
+		return apply_filters( 'wl_mappings_register_mappings', $mappings );
+
 	}
 
 	/**
 	 * Returns a list of property rows
 	 *
-	 * @param int $mapping_id Primary key of mapping table.
+	 * @param string $mapping_id Primary key of mapping table.
 	 *
 	 * @return array List of property items belong to $mapping_id.
 	 */
 	public function get_properties( $mapping_id ) {
-		$property_table_name = $this->wpdb->prefix . WL_PROPERTY_TABLE_NAME;
-		$property_rows       = $this->wpdb->get_results(
-			$this->wpdb->prepare( "SELECT * FROM $property_table_name WHERE mapping_id=%d", $mapping_id ),
-			ARRAY_A
-		);
-
-		return $property_rows;
+		$property_table_name   = $this->wpdb->prefix . WL_PROPERTY_TABLE_NAME;
+		$ui_defined_properties = array();
+		if ( is_numeric($mapping_id) ) {
```

Comment:
@naveen17797 pls add comments here explaining what's happening: what's the difference between numeric and non numeric mappings IDs?